drivers/ethernet/eth_native_posix.c

```diff
@@ -323,9 +454,92 @@ static const struct ethernet_api eth_if_api = {
 #if defined(CONFIG_NET_STATISTICS_ETHERNET)
 	.stats = &eth_context_data.stats,
 #endif
+#if defined(CONFIG_ETH_NATIVE_POSIX_PTP_CLOCK)
+	.get_ptp_clock = eth_get_ptp_clock,
+#endif
 };
 
 ETH_NET_DEVICE_INIT(eth_native_posix, CONFIG_ETH_NATIVE_POSIX_DRV_NAME,
 		    eth_init, &eth_context_data, NULL,
 		    CONFIG_KERNEL_INIT_PRIORITY_DEFAULT, &eth_if_api,
 		    _ETH_MTU);
+
+#if defined(CONFIG_ETH_NATIVE_POSIX_PTP_CLOCK)
+struct ptp_context {
+	struct eth_context *eth_context;
+};
+
+static struct ptp_context ptp_0_context;
+
+
+static int ptp_clock_set_native_posix(struct device *clk,
+				      struct net_ptp_time *tm)
+{
+	ARG_UNUSED(clk);
+	ARG_UNUSED(tm);
+
+	/* We cannot set the host device time so this function
+	 * does nothing.
+	 */
```

Comment:
Would you want to be able to?

Reply:
Setting the host device time is probably not needed here, at least for now. The reason being that the native_posix target is typically used when testing things and it might not be ok to change the host time in this case. Of course we could make this optional if someone really wishes to have this kind of feature. I am ok if you send a PR that adds this kind of support thou.

Reply:
I misunderstood you then. That kind of fake time would be much welcomed feature so if you have time to add such a feature, I would appreciate that.

Reply:
Roger. I added a note in #6044 about it. Hopefully I will have in the following days to add it.